core/search/disjunction.hpp

```diff
@@ -1433,56 +1300,34 @@ class block_disjunction final : public doc_iterator, private score_ctx {
   uint64_t* begin_{std::end(mask_)};
   uint64_t cur_{};
   doc_id_t doc_base_{doc_limits::invalid()};
-  doc_id_t min_{doc_limits::min()}; // base doc id for the next mask
-  doc_id_t max_{doc_limits::invalid()}; // max doc id in the current mask
+  doc_id_t min_{doc_limits::min()};      // base doc id for the next mask
+  doc_id_t max_{doc_limits::invalid()};  // max doc id in the current mask
   attributes attrs_;
   size_t match_count_;
-  size_t buf_offset_{}; // offset within a buffer
-  score_buffer_type score_buf_;     // FIXME EBO
-  min_match_buffer_type match_buf_; // FIXME EBO
-  const byte_type* score_value_{score_buf_.data()};
-  order::prepared::merger merger_;
-}; // block_disjunction
-
-template<
-  typename DocIterator,
-  typename Adapter = score_iterator_adapter<DocIterator>>
-using scored_disjunction_iterator = block_disjunction<
-  DocIterator,
-  block_disjunction_traits<true, MatchType::MATCH, false>,
-  Adapter>;
-
-template<
-  typename DocIterator,
-  typename Adapter = score_iterator_adapter<DocIterator>>
-using disjunction_iterator = block_disjunction<
-  DocIterator,
-  block_disjunction_traits<false, MatchType::MATCH, false>,
-  Adapter>;
-
-template<
-  typename DocIterator,
-  typename Adapter = score_iterator_adapter<DocIterator>>
-using scored_min_match_iterator = block_disjunction<
-  DocIterator,
-  block_disjunction_traits<true, MatchType::MIN_MATCH, false>,
-  Adapter>;
-
-template<
-  typename DocIterator,
-  typename Adapter = score_iterator_adapter<DocIterator>>
-using min_match_iterator = block_disjunction<
-  DocIterator,
-  block_disjunction_traits<false, MatchType::MIN_MATCH, false>,
-  Adapter>;
-
-//////////////////////////////////////////////////////////////////////////////
-/// @returns disjunction iterator created from the specified sub iterators
-//////////////////////////////////////////////////////////////////////////////
+  size_t buf_offset_{};              // offset within a buffer
+  score_buffer_type score_buf_;      // FIXME EBO
+  min_match_buffer_type match_buf_;  // FIXME EBO
```

Comment:
Just use [[no_unique_address]]
It already should be supported by all our compilers